app/src/main/java/net/bible/service/db/readingplan/ReadingPlanDBAdapter.kt

```diff
@@ -0,0 +1,767 @@
+import org.crosswire.jsword.passage.Key
+import org.crosswire.jsword.versification.Versification
+import org.crosswire.jsword.versification.system.SystemNRSVA
+import org.crosswire.jsword.versification.system.Versifications
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.Date
+import kotlin.collections.ArrayList
+
+/**
+ * @author Timmy Braun [tim.bze at gmail dot com] (2/14/2019)
+ */
+class ReadingPlanDBAdapter {
+    companion object {
+        const val TAG = "ReadingPlanDBAdapter"
+        const val INCLUSIVE_VERSIFICATION = SystemNRSVA.V11N_NAME
+        val app = BibleApplication.application
+        val dbHelper: CommonDatabaseHelper = CommonDatabaseHelper.getInstance()
+
+        private val dbWritable = dbHelper.writableDatabase
+        private val dbReadable = dbHelper.readableDatabase
+
+        private val dateBasedFormatMonthDay = SimpleDateFormat("MMM-d")
+        val dateBasedFormatWithYear = SimpleDateFormat("MMM-d/yyyy")
+
+    }
+
+    var metaCurrentActiveReadingPlanID: Int?
+        get() {
+            val projection = arrayOf(BaseColumns._ID)
+            val selection = "${ReadingPlanMeta.COLUMN_LAST_USED_PLAN}=?"
+            val selectionArgs = arrayOf(DB_TRUE_VALUE)
+            val cursor = dbReadable.query(
+                ReadingPlanMeta.TABLE_NAME,
+                projection,
+                selection,
+                selectionArgs,
+                null,
+                null,
+                null)
+            var result: Int? = null
+            if (cursor.moveToFirst()) {
+                val readID: Int = cursor.getInt(0)
+                cursor.close()
+                result = readID
+            }
+            cursor.close()
+            return result
+        }
+        set(ReadPlanID) {
+            if (ReadPlanID != null && ReadPlanID != 0) {
+                val setFalseValues = ContentValues().apply {
+                    put(ReadingPlanMeta.COLUMN_LAST_USED_PLAN, DB_FALSE_VALUE)
+                }
+                val setTheTrueValue = ContentValues().apply {
+                    put(ReadingPlanMeta.COLUMN_LAST_USED_PLAN, DB_TRUE_VALUE)
+                }
+                val whereClause = "${ReadingPlanMeta.COLUMN_ID}=?"
+                val whereArgs = arrayOf(ReadPlanID.toString())
+                // Set all plans to false. ONLY 1 plan is the Current Active Plan
+                dbWritable.update(
+                    ReadingPlanMeta.TABLE_NAME,
+                    setFalseValues,
+                    null,
+                    null)
+                // Set the 1 Current Plan to Active
+                dbWritable.update(
+                    ReadingPlanMeta.TABLE_NAME,
+                    setTheTrueValue,
+                    whereClause,
+                    whereArgs)
+            }
+        }
+
+    val getCurrentPlanShortCode: String =
+        StringUtils.left(getMetaFileNameFromId(metaCurrentActiveReadingPlanID ?: 0),8)
```

Comment:
It looks like you are trying to make a property here too. Now this is constant value (initialized when instance is created). Whether constant or property, please omit get prefix.

Reply:
Done